packages/react/README.md

```diff
@@ -30,12 +30,60 @@ Sentry.init({
 
 // ...
 
-ReactDOM.render(<App />, rootNode);
+const container = document.getElementById(“app”);
+const root = createRoot(container);
+root.render(<App />);
 
-// Can also use with React Concurrent Mode
-// ReactDOM.createRoot(rootNode).render(<App />);
+// also works with hydrateRoot
+// const domNode = document.getElementById('root');
+// const root = hydrateRoot(domNode, reactNode);
+// root.render(<App />);
 ```
 
+### React 19
+
+Starting with React 19, the `createRoot` and `hydrateRoot` methods expose error hooks that can be used to capture errors automatically. Use the `Sentry.captureReactException` method to capture errors in the error hooks you are interested in.
+
+```js
+const container = document.getElementById(“app”);
+const root = createRoot(container, {
+  // Callback called when an error is thrown and not caught by an Error Boundary.
+  onUncaughtError: (error, errorInfo) => {
+    Sentry.captureReactException(error, errorInfo);
```

Comment:
I would vote for a different name here - imho this is a bit misleading, like I could easily see users just using this instead of `captureException()` 🤔 

What if instead we'd do something like this:

```js
const root = createRoot(container, {
  // Callback called when an error is thrown and not caught by an Error Boundary.
  onUncaughtError: Sentry.reactErrorHandler(),
  onCaughtError: Sentry.reactErrorHandler((error, errorInfo) => {
    // optional callback if users want more/custom config in addition?
  })
});
```

Reply:
I like `Sentry.reactErrorHandler` a lot better - will change!

Reply:
I have some concerns with my initial approach that also applies to `Sentry.reactErrorHandler`. https://github.com/getsentry/sentry-javascript/issues/11798#issuecomment-2124667149

Pinged react team member on this, let's see if we can get feedback.